src/app/content/components/TableOfContents/index.tsx

```diff
@@ -150,51 +162,131 @@ function maybeAriaLabel(page: LinkedArchiveTreeSection) {
     return {};
   }
 
-  return {'aria-label': `${titleText} - Chapter ${parentNum}`};
+  return { 'aria-label': `${titleText} - Chapter ${parentNum}` };
+}
+
+function ArchiveTreeComponent({
+  item,
+  book,
+  page,
+  activeSection,
+  onNavigate,
+  onKeyDown,
+}: {
+  item: LinkedArchiveTree;
+  book: Book | undefined;
+  page: Page | undefined;
+  activeSection: React.RefObject<HTMLElement>;
+  onNavigate: () => void;
+  onKeyDown: (props: KeyboardSupportProps) => void;
+}) {
+  const sectionType = getArchiveTreeSectionType(item);
+
+  const [isOpen, setOpen] = React.useState<boolean>(shouldBeOpen(page, item));
+
+  const toggleOpen = () => {
+    setOpen((prevState) => !prevState);
+  };
+
+  const onKeyDownSupport = (e: React.KeyboardEvent<HTMLAnchorElement>) => {
+    onKeyDown({
+      event: e,
+      item,
+      isOpen,
+      onSelect: toggleOpen,
+    });
+  };
+
+  return (
+    <Styled.NavItem key={item.id} sectionType={sectionType}>
+      <TocNode
```

Comment:
as i mentioned in the `details` comment, a better name for this at this point might be `TocSectionToggle`